File: website/src/model/message.ts
import { BaseQuery, User } from '@/model/common'
export interface Message {
  id: number
  title: string
  content: string
  send_by: number
  send_by_user?: User
  send_to: number
  send_to_user?: User
  status: number
  created_at?: string
  updated_at?: string
}

export type MessageQuery = BaseQuery & {
  self?: number
  field_eq_send_by?: number
  field_lk_title?: string
  field_lk_content?: string
  field_eq_status?: number | null
}

export enum MessageStatus {
  UNREAD = 1,
  HASREAD = 2
}

export interface MessageTemplate {
  id: number
  name: string
  title: string
  content: string
  create_by: number
  create_by_user?: User
  created_at?: string
  updated_at?: string
}

export type MessageTemplateQuery = BaseQuery & {
  field_lk_name: string
  field_lk_title: string
  field_lk_content: string
}

File: databases/model/message.go
// Package model 消息通知数据库模型
package model

const (
	UnRead  = 1
	HasRead = 2
)

type Message struct {
	ID         int       `json:"id" gorm:"column:id;primary_key"` // 消息编号
	Title      string    `json:"title" gorm:"column:title"`       // 消息标题
	Content    string    `json:"content" gorm:"column:content"`   // 消息内容
	SendBy     int       `json:"send_by" gorm:"column:send_by"`   // 消息发送人
	SendByUser *User     `json:"send_by_user,omitempty" gorm:"<-:false;ForeignKey:SendBy;AssociationForeignKey:ID"`
	SendTo     int       `json:"send_to" gorm:"column:send_to"` // 消息接收者
	SendToUser *User     `json:"send_to_user,omitempty" gorm:"<-:false;ForeignKey:SendTo;AssociationForeignKey:ID"`
	Status     int       `json:"status" gorm:"column:status"`         // 消息状态
	CreatedAt  Timestamp `json:"created_at" gorm:"column:created_at"` // 消息发送时间
	DeletedAt  Timestamp `json:"deleted_at" gorm:"column:deleted_at"` // 消息删除时间
	UpdatedAt  Timestamp `json:"updated_at" gorm:"column:updated_at"` // 消息更新时间

}

func (Message) TableName() string {
	return "message"
}
